android/brave_java_sources.gni

```diff
@@ -375,7 +385,41 @@ brave_java_base_sources = [
 brave_java_base_module_sources = [
   "//brave/android/java/org/chromium/chrome/browser/base/SplitCompatJobIntentService.java",
   "//brave/android/java/org/chromium/chrome/browser/upgrade/BraveUpgradeJobIntentService.java",
+  "//brave/android/java/org/chromium/chrome/browser/vpn/wireguard/WireguardService.java",
 ]
 
-brave_java_base_module_deps =
-    [ "//third_party/androidx:androidx_core_core_java" ]
+brave_java_base_module_deps = [
+  "//brave/third_party/android_deps:com_wireguard_android_java",
+  "//third_party/androidx:androidx_core_core_java",
+]
+
+if (is_android) {
+  brave_wireguard_deps =
+      [ "//brave/third_party/android_deps:com_wireguard_android_J__unpack_aar" ]
+
+  _lib_dir = get_label_info(
+                 "//brave/third_party/android_deps:com_wireguard_android($default_toolchain)",
+                 "target_out_dir") + "/com_wireguard_android_java/jni"
+
+  if (target_cpu == "arm64") {
+    primary_abi = "arm64-v8a"
+    secondary_abi = "armeabi-v7a"
+  } else if (target_cpu == "arm") {
+    primary_abi = "armeabi-v7a"
+    secondary_abi = "arm64-v8a"
+  } else if (target_cpu == "x64") {
+    primary_abi = "x86_64"
+    secondary_abi = "x86"
+  } else if (target_cpu == "x86") {
+    primary_abi = "x86"
+    secondary_abi = "x86_64"
+  } else {
+    assert(false, "Unknown Android ABI: " + current_cpu)
+  }
```

Comment:
is this block necessary? why we cannot use vars from `build/config/android/abi.gni`?